app/app.model.js

```diff
@@ -1,5 +1,24 @@
 const db = require("../db/connection")
 
+const selectArticles = () => {
+    return db.query(`
+        SELECT articles.*, 
+        COALESCE (comment_count.count :: INT, 0) AS comment_count
+        FROM articles
+        LEFT OUTER JOIN
+        (
+        SELECT article_id, COUNT(*)
+        FROM comments
+        GROUP BY article_id
+        ) comment_count
+        ON articles.article_id = comment_count.article_id
+        ORDER BY articles.created_at DESC
+        `)
```

Comment:
This is definetly an over engineered solution, you dont need a coalesce here just using the COUNT(*) should get you to the answer. I would recommend refactoring this as its very hard to read and understand.